### PR TITLE
Refactor: Separate master and PR workflows

### PR DESCRIPTION
**Pull request by emranbm**

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+name: Pull Request Workflow
+on:
+  pull_request:
+    branches:
+      - master
+jobs:
+  test:
```
Comment (Adibov) [inline .github/workflows/pr.yml:8]: Do we need duplication of this job? If so, can't we write a template, and use it multiple times instead?

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '1.19.2'
+      - name: Test
+        run: go test ./... -v -coverprofile=coverage.txt -covermode count
+  build:
```
Comment (Adibov) [inline .github/workflows/pr.yml:17]: Same thing about this job.

**Review by Adibov — APPROVED** [review]

File: .github/workflows/pr.yml (hunk 1 above)
Comment (emranbm) [reply .github/workflows/pr.yml:8]: Good point. But, reusing jobs is not as smooth as in GitLab. Especially, these jobs are not fully the same, and handling non-duplicate code increases complexity, which IMO isn't worth it.

Comment (emranbm) [reply .github/workflows/pr.yml:8]: Links:
1. Struggles about supporting code reuse in GitHub Actions: [link](https://github.com/actions/starter-workflows/issues/245)
2. Current solution for action-level code reuse: [link](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action)

Comment (Adibov) [reply .github/workflows/pr.yml:8]: Nice one, thanks for following up.

**Pull request merged**